library/core/src/ptr/non_null.rs

```diff
@@ -205,6 +240,22 @@ impl<T: ?Sized> NonNull<T> {
     }
 
     /// Acquires the underlying `*mut` pointer.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::ptr::NonNull;
+    ///
+    /// let mut x = 0u32;
+    /// let ptr = NonNull::<u32>::new(&mut x as *mut _).expect("ptr is null!");
```

Comment:
the turbofish and `as` cast seems unnecessary, is this intended?

```rust
let ptr = NonNull::new(&mut x).expect("ptr is null!");
```

Reply:
You're absolutely right! Removing the turbofish.

Reply:
@GuillaumeGomez  well change the remaining examples then 😅 r=me after that.

Reply:
Sorry, mornings...